packages/wonder-blocks-dropdown/components/dropdown-launcher.js

```diff
@@ -0,0 +1,244 @@
+            onChange([...selectedValues, selectedValue]);
+        }
+        this.handleItemSelected();
+    };
+
+    _openDropdown = () => {
+        this.setState({open: true});
+    };
+
+    _getMenuItems = (): Array<DropdownItem> => {
+        const {selectedValues} = this.props;
+        return React.Children.toArray(this.props.menuItems)
+            .filter(Boolean)
+            .map((item) => {
+                const {value, disabled} = item.props;
+                const itemObject = {
+                    component: item,
+                    focusable:
+                        ActionItem.isClassOf(item) || OptionItem.isClassOf(item)
+                            ? !disabled
+                            : false,
+                    populatedProps: {},
+                };
+                if (ActionItem.isClassOf(item)) {
+                    return {
+                        ...itemObject,
+                        populatedProps: {
+                            indent: true,
+                            onClick: this.handleItemSelected,
+                        },
+                    };
+                } else if (OptionItem.isClassOf(item)) {
+                    return {
+                        ...itemObject,
+                        populatedProps: {
+                            onToggle: this.handleOptionSelected,
+                            selected: selectedValues
+                                ? selectedValues.includes(value)
+                                : false,
+                            variant: "check",
+                        },
+                    };
+                } else {
+                    return itemObject;
+                }
+            });
+    };
+
+    handleOpenerRef = (node: any) => {
+        this.openerElement = ((ReactDOM.findDOMNode(node): any): HTMLElement);
+    };
+
+    render() {
+        const dropdownItems = this._getMenuItems();
+        const {
+            children,
+            alignment,
+            disabled,
+            style,
+            dropdownStyle,
+            /* eslint-disable no-unused-vars */
+            onChange,
+            selectedValues,
+            "aria-disabled": ariaDisabled,
+            ...sharedProps
+        } = this.props;
+
+        const opener = (
+            <span
+                ref={this.handleOpenerRef}
+                disabled={!dropdownItems.length || disabled}
+            >
+                {children(this._openDropdown)}
+            </span>
+        );
+
+        return (
+            <Dropdown
+                alignment={alignment}
+                items={this._getMenuItems()}
+                keyboard={this.state.keyboard}
+                dropdownStyle={[styles.menuTopSpace, dropdownStyle]}
+                onOpenChanged={this.handleOpenChanged}
+                open={this.state.open}
+                opener={opener}
+                openerElement={this.openerElement}
+                role="menu"
+                style={style}
```

Comment:
Does the `style` prop work in the example?  Can you set `marginLeft: 100` to see if applies the style correctly?

Reply:
It works in some cases, marginLeft isnt working, but menuTopSpace is working

Reply:
I don't think we need this anymore after the addition of `DropdownAnchor`.  People can set `style` on the anchor itself.